mipengine/controller/node_landscape_aggregator.py

```diff
@@ -305,6 +307,25 @@ def _get_datasets_per_node(
     return datasets_per_node
 
 
+def remove_duplicated_datasets(datasets_per_node):
```

Comment:
This method should remove datasets that are duplicated inside a data model, not across the federation.